src/main/java/org/kohsuke/github/extras/authorization/JwtBuilderUtil.java

```diff
@@ -0,0 +1,151 @@
+package org.kohsuke.github.extras.authorization;
+
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.io.Serializer;
+import io.jsonwebtoken.jackson.io.JacksonSerializer;
+import io.jsonwebtoken.security.SignatureAlgorithm;
+import org.kohsuke.github.GHException;
+
+import java.lang.reflect.Method;
+import java.security.Key;
+import java.security.PrivateKey;
+import java.time.Instant;
+import java.util.Date;
+import java.util.logging.Logger;
+
+/**
+ * This is a util to build a JWT.
+ *
+ * <p>
+ * This class is used to build a JWT using the jjwt library. It uses reflection to support older versions of jjwt. The
+ * class may be removed once we are sure we no longer need to support pre-0.12.x versions of jjwt.
+ * </p>
+ */
+final class JwtBuilderUtil {
+
+    private static final Logger LOGGER = Logger.getLogger(JwtBuilderUtil.class.getName());
+
+    /**
+     * Build a JWT.
+     *
+     * @param issuedAt
+     *            issued at
+     * @param expiration
+     *            expiration
+     * @param applicationId
+     *            application id
+     * @param privateKey
+     *            private key
+     * @return JWT
+     */
+    static String buildJwt(Instant issuedAt, Instant expiration, String applicationId, PrivateKey privateKey) {
+
+        try {
+            return DefaultBuilderImpl.buildJwt(issuedAt, expiration, applicationId, privateKey);
+        } catch (NoSuchMethodError | NoClassDefFoundError e) {
+            LOGGER.info(
```

Comment:
Changed this from warning to info to keep from spamming logs too heavily.  What we really want is one `warning` when the method first switches to reflective building and then not again.